File: youtube-links/src/shared/features/users/commands/ConfirmEmail.ts
import { ValidationConsts } from '@/shared/localization/ValidationConsts';

export namespace ConfirmEmail {
  export interface Command {
    email: string;
    token: string;
  }

  export const Validation = {
    email: {
      notEmpty: (v: string) => !!v || 'Email should not be empty.',
      maximumLength: (v: string) =>
        v.length <= ValidationConsts.MaximumStringLength ||
        `The length of email must be ${ValidationConsts.MaximumStringLength} characters or fewer. You entered ${v.length} characters.`,
      isEmailAddress: (v: string) =>
        ValidationConsts.isValidEmail(v) || 'Email address is not valid.',
    },
    token: {
      notEmpty: (v: string) => !!v || 'Token should not be empty.',
    },
  };
}

File: YoutubeLinks.Shared/Features/Users/Commands/ConfirmEmail.cs
using FluentValidation;
using MediatR;
using Microsoft.Extensions.Localization;
using YoutubeLinks.Shared.Localization;

namespace YoutubeLinks.Shared.Features.Users.Commands
{
    public static class ConfirmEmail
    {
        public class Command : IRequest<bool>
        {
            public string Email { get; set; }
            public string Token { get; set; }
        }

        public class Validator : AbstractValidator<Command>
        {
            public Validator(IStringLocalizer<ValidationMessage> localizer)
            {
                RuleFor(x => x.Email)
                    .NotEmpty()
                    .WithMessage(x => localizer[nameof(ValidationMessageString.EmailNotEmpty)])
                    .MaximumLength(ValidationConsts.MaximumStringLength)
                    .WithMessage(x => localizer[nameof(ValidationMessageString.EmailMaximumLength)])
                    .EmailAddress()
                    .WithMessage(x => localizer[nameof(ValidationMessageString.EmailIsEmailAddress)]);

                RuleFor(x => x.Token)
                    .NotEmpty()
                    .WithMessage(x => localizer[nameof(ValidationMessageString.TokenNotEmpty)]);
            }
        }
    }
}
